src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -133,17 +136,31 @@ internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, Ca
             }
         }
 
-        private async Task<IEslintBridgeClient> GetJavaScriptClient(CancellationToken token)
+        private async Task<IEslintBridgeClient> GetJavaScriptClient(CancellationToken cancellationToken)
         {
             var port = await eslintBridgeProcess.Start();
 
-            if (port != javascriptServerPort)
+            try
             {
-                javascriptServerPort = port;
-                javascriptClient?.Dispose();
-                javascriptClient = eslintBridgeClientFactory.Create(javascriptServerPort);
+                serverInitLocker.WaitOne();
```

Comment:
I'm breaking my head trying to find a way to test this.
I wrote a test that launches two tasks with two EventWaitHandles that simulate a hang in the `InitLinter`.
However -- my tests still pass even when the second task hasn't even started yet. I'm trying to find a way to make sure that the second task is actually blocked, but I can't think of anything. Some stackoverflow posts recommend adding an internal var "waitingThreadsCount", but I don't want to pollute the product code for this.
I've tried using the `eslintBridgeProcess.Start` but again, the tests pass even when the second task hasn't reached the "WaitOne" block. 
Got any ideas? 🤔

Reply:
> I guess in a way it doesn't make sense to test this, because it's an equivalent of attempting to test a lock statement, just with an async call.

Well, we want to test how our code uses the lock statement rather than the lock statement itself i.e. whether the lock guarding the correct section of code.
That doesn't it's possible/easy to test though ;-)

Moving the call to Start inside the lock might make it easier to test.

Reply:
Actually, I'm not sure it will be any easier test e.g.

Task 1 - start analysis 1
 - block thread in Start call
Task 2 - start analysis 2

Task 2 should be blocked until Task 1 has finished.
- wait for Task2 for a short timeout
- check the wait timed out
- check Start hasn't been called twice

- unblock Start call and check Task 2 runs to completion.

Ugly, and not that different from using a Sleep.

Can you think of anything neater?

Reply:
No :( and in the end it still relies on the Sleep. It's very similar to the test that I wrote, but without the sleep, there's no way to really test the behavior. And I don't really want to use a sleep...

Reply:
Yep, the only way to test the locking is either to use a sleep-like approach and test if from the outside, or to put in more test hooks so that the behaviour can be controlled non-deterministically. Or not test it at all.